baremetal/metal3machine_manager.go

```diff
@@ -1633,6 +1633,8 @@ func (m *MachineManager) WaitForM3Metadata(ctx context.Context) error {
 
 	// If it is not ready yet, wait.
 	if !metal3Data.Status.Ready {
+		m.Log.Info("Waiting for Metal3Data to be ready")
+		m.SetConditionMetal3MachineToFalse(infrav1.KubernetesNodeReadyCondition, infrav1.WaitingforMetal3DataReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
I it would make sense here if we add new condition like  https://github.com/metal3-io/cluster-api-provider-metal3/pull/836/files#diff-cd44a3d05ab777f57cbb6377e9ad6aa4e783aace2741f72c89b76aa4d540daccR55

Reply:
This part is handled in your PR